android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/translate/TranslationEngineState.kt

```diff
@@ -19,3 +19,81 @@ data class TranslationEngineState(
     val isEngineReady: Boolean? = false,
     val requestedTranslationPair: TranslationPair? = null,
 )
+
+/**
+ * Determines the best initial "to" language based on the translation state and user preferred
+ * languages.
+ *
+ * @param candidateLanguages The language options available to select as a final initial value.
+ * @return The best determined "to" language or null if a determination cannot be made.
+ */
+fun TranslationEngineState.initialToLanguage(candidateLanguages: List<Language>?): Language? {
+    return determineInitialDropdownValue(
+        translationEngineState = this,
+        isFromLanguage = false,
+        candidateLanguages = candidateLanguages,
+    )
+}
```

Comment:
IMO it would be more clear to handle each use case in each function instead of combining them in determineInitialDropdownValue
```suggestion
fun TranslationEngineState.initialToLanguage(candidateLanguages: List<Language>?): Language? = 
    candidateLanguages?.find { 
        it.code == (requestedTranslationPair?.toLanguage ?: detectedLanguages?.userPreferredLangTag) 
    }
```

Reply:
Updated, ty! Thanks for suggesting the idiomatic kotlin format too